### ISSUE TITLE
Help with usage with nested addon

### ISSUE THREAD

**Issue by samchrisinger** [dfreeman/ember-cli-node-assets#1]
I'm building an Ember addon that has [ember-data-url-templates](https://github.com/amiel/ember-data-url-templates) as a dependency. That addon uses ember-cli-node-assets to import the uri-templates.js library like: https://github.com/amiel/ember-data-url-templates/blob/master/index.js. 

This works great when installing ember-data-url-templates from an ember app, but I'm having trouble when trying to nest addons-- basically the uri-templates.js module never seems to be imported. From the node debugger, I _do_ see that the uri-template module is added to a build tree somewhere (sorry for vagueness, the backtrace is a bit hard to decipher). 

Basically, before I spend way too much time debugging this, can you offer any thoughts on whether or not this module is expected to work for nested addons or advice on where to look?

**Comment by dfreeman** [comment]
By design, this addon doesn't import dependencies unless the project it's acting on behalf of is either an app or an addon directly required by an app. If two different addons, for instance, both depended on `ember-data-url-templates`, you could end up with two copies of the underlying `uri-templates.js` library in your vendor file, potentially leading to name conflicts, undefined behavior, cats and dogs getting along, and other general mayhem :).

Currently the simplest answer is to make `ember-data-url-templates` a `peerDependency` of your addon, the same way [`ember-tether` suggests](https://github.com/yapplabs/ember-tether#using-ember-tether-in-your-own-addon) when developing an addon that makes use of it.

**Comment by dfreeman** [comment]
Also of note: there are [early signs](https://github.com/ember-cli/ember-cli/pull/5877) of some changes to how transitive dependencies are dealt with in Ember CLI—depending on how that evolves, the situation here may change. For now, though, `peerDependencies` is probably your best bet.

**Comment by samchrisinger** [comment]
Thanks so much for the explanation and suggestions!
